http/src/request/template/update_template.rs

```diff
@@ -0,0 +1,127 @@
+use crate::request::prelude::*;
+use std::{
+    error::Error,
+    fmt::{Display, Formatter, Result as FmtResult},
+};
+use twilight_model::{id::GuildId, template::Template};
+
+#[derive(Clone, Debug)]
+#[non_exhaustive]
+pub enum UpdateTemplateError {
+    /// Name of the template is invalid.
+    NameInvalid {
+        /// Provided name.
+        name: String,
+    },
+    /// Description of the template is invalid.
+    DescriptionTooLarge {
+        /// Provided description.
+        description: String,
+    },
+}
+
+impl Display for UpdateTemplateError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        match self {
+            Self::NameInvalid { .. } => f.write_str("the template name is invalid"),
+            Self::DescriptionTooLarge { .. } => {
+                f.write_str("the template description is too large")
+            }
+        }
+    }
+}
+
+impl Error for UpdateTemplateError {}
+
+#[derive(Serialize)]
+struct UpdateTemplateFields {
+    name: Option<String>,
+    description: Option<String>,
+}
+
+/// Update the template's metadata, by ID and code.
+///
+/// # Errors
```

Comment:
These errors shouldn't be on the struct itself, but they're more useful on the methods themselves